src/renderer/src/composables/Miscellaneous/useIPCListeners.ts

```diff
@@ -33,6 +37,18 @@ export function useIPCListeners() {
             },
         );
 
+        window.electron.ipcRenderer.on(
+            'schemaOpened',
+            async (_, file: Uint8Array) => {
+                if (!vueFlow) return;
+                historyStore.$reset();
+                fileStore.$reset();
+                const contents = await importHelper(file);
+                const [nodes, edges] = contents;
+                createNodeFromImport(nodes), createEdgeFromImport(edges, nodes);
+            },
```

Comment:
You can call the `autoLayout()` function here, which is located in the 'useNodeAutoLayout' composable file. Please note that you should use async/await to achieve this. This ensures that Vue renders all of the nodes first without concerning how they appear on the canvas. Once the rendering process is complete, you can then execute `autoLayout()`.


Something like this:
```ts
async (_, file: Uint8Array) => {
    if (!vueFlow) return;
    historyStore.$reset();
    fileStore.$reset();
    const contents = await importHelper(file);
    const [nodes, edges] = contents;
    createNodeFromImport(nodes), createEdgeFromImport(edges, nodes);
    
    await nextTick();
    runAutoLayout();
},
```

Reply:
autoLayout in the useIRCListener file will cause a Pinia Error "... pinia is called before initialized"

Reply:
May I see a code snippet of how you did it?